pkg/network/setup/netstat.go

```diff
@@ -0,0 +1,280 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+type NetStat struct {
+	ifaceCacheFactory cache.InterfaceCacheFactory
+
+	// In memory cache, storing pod interface information.
+	// key is the file path, value is the contents.
+	// if key exists, then don't read directly from file.
+	podInterfaceVolatileCache PodInterfaceByVMIAndName
+}
+
+func NewNetStat(ifaceCacheFactory cache.InterfaceCacheFactory) *NetStat {
+	return &NetStat{
+		ifaceCacheFactory:         ifaceCacheFactory,
+		podInterfaceVolatileCache: PodInterfaceByVMIAndName{},
+	}
+}
+
+func (c *NetStat) Teardown(vmi *v1.VirtualMachineInstance) {
+	c.podInterfaceVolatileCache.DeleteAllForVMI(vmi.UID)
+}
+
+func (c *NetStat) PodInterfaceVolatileDataIsCached(vmi *v1.VirtualMachineInstance, ifaceName string) bool {
+	_, exists := c.podInterfaceVolatileCache.Load(vmi.UID, ifaceName)
+	return exists
+}
+
+func (c *NetStat) CachePodInterfaceVolatileData(vmi *v1.VirtualMachineInstance, ifaceName string, data *cache.PodCacheInterface) {
+	c.podInterfaceVolatileCache.Store(vmi.UID, ifaceName, data)
+}
+
+func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domain) error {
+	if domain == nil {
+		return nil
+	}
+
+	// This is needed to be backwards compatible with vmi's which have status interfaces
+	// with the name not being set
+	if len(domain.Spec.Devices.Interfaces) == 0 && len(vmi.Status.Interfaces) == 1 && vmi.Status.Interfaces[0].Name == "" {
+		for _, network := range vmi.Spec.Networks {
+			if network.NetworkSource.Pod != nil {
+				vmi.Status.Interfaces[0].Name = network.Name
+			}
+		}
+	}
+
+	if len(vmi.Status.Interfaces) == 0 {
+		// Set Pod Interface
+		interfaces := make([]v1.VirtualMachineInstanceNetworkInterface, 0)
+		for _, network := range vmi.Spec.Networks {
+			podIface, err := c.getPodInterfacefromFileCache(vmi, network.Name)
+			if err != nil {
+				return err
+			}
+
+			if podIface != nil {
+				ifc := v1.VirtualMachineInstanceNetworkInterface{
```

Comment:
Is it possible to please change the "ifc" variable name to something more descriptive?

Reply:
Thanks for your clarification.